crates/orchestrator/src/tests/workers/mod.rs

```diff
@@ -0,0 +1,85 @@
+use crate::config::config_force_init;
+use crate::database::MockDatabase;
+use crate::jobs::types::{ExternalId, JobItem, JobStatus, JobType};
+use crate::queue::job_queue::JobQueueMessage;
+use crate::queue::MockQueueProvider;
+use crate::tests::common::init_config;
+use crate::workers::snos::SnosWorker;
+use crate::workers::Worker;
+use da_client_interface::MockDaClient;
+use httpmock::MockServer;
+use rstest::rstest;
+use serde_json::json;
+use std::collections::HashMap;
+use uuid::Uuid;
+
+#[rstest]
+#[tokio::test]
+async fn test_create_job() {
+    let server = MockServer::start();
+    let da_client = MockDaClient::new();
+    let mut db = MockDatabase::new();
+    let mut queue = MockQueueProvider::new();
+
+    // Mocking db functions
+    db.expect_get_latest_job_by_type().returning(|_| Ok(None)).call(JobType::SnosRun).expect("Failed to call.");
```

Comment:
should we also do a test case for when the db returns a value? we could use `rstest` with `cases` so that we can use the same code only